lib/typed/coercion/coercer_registry.rb

```diff
@@ -41,11 +42,20 @@ def register(coercer)
       sig { void }
       def reset!
         @available = DEFAULT_COERCERS.clone
+        @coercer_cache.clear
       end
 
-      sig { params(type: T::Types::Base).returns(T.nilable(T.class_of(Coercer))) }
+      sig { params(type: T::Types::Base).returns(T.nilable(Coercer)) }
       def select_coercer_by(type:)
-        @available.find { |coercer| coercer.new.used_for_type?(type) }
+        cache_key = type.to_s
+        if @coercer_cache.key?(cache_key)
+          return @coercer_cache[cache_key]
+        end
+
+        klass = @available.find { |coercer| coercer.new.used_for_type?(type) }
+        instance = klass&.new
+        @coercer_cache[cache_key] = instance if instance
+        instance
```

Comment:
nitpick: this is personal preference, but I think early returning on the `nil` here is a bit more clear and avoids a single variable return (assignment will return what was assigned)


```suggestion
        return unless klass
        
        @coercer_cache[cache_key] = klass.new
```

Reply:
Updated with https://github.com/maxveldink/sorbet-schema/pull/132/commits/d463b5833626e489d3350358008b39bd4ae70270!